frontend/src/components/Sidebar/MapComponents/CustomControl.tsx

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+
+// Third-party
+import * as am5map from '@amcharts/amcharts5/map';
+import {ZoomIn, ZoomOut, ZoomOutMap} from '@mui/icons-material';
+import {ButtonGroup, IconButton, Stack} from '@mui/material';
+
+interface CustomControlProps {
+  chart: am5map.MapChart | null;
+  onHomeClick?: () => void;
+  maxZoomLevel: number; // MaxZoom Var from HeatMap.tsx
+}
+
+export default function CustomControl({chart, onHomeClick, maxZoomLevel}: CustomControlProps) {
+  // Zoom In Handler
+  const handleZoomIn = () => {
+    if (chart && chart.seriesContainer) {
+      try {
+        const currentZoom = chart.get('zoomLevel', 1);
+        const newZoom = Math.min(currentZoom * 2, maxZoomLevel);
+
+        const centerX = chart.seriesContainer.width() / 2;
+        const centerY = chart.seriesContainer.height() / 2;
+
+        // zoom to new Point with animation (300ms)
+        chart.zoomToPoint({x: centerX, y: centerY}, newZoom, true, 300);
+      } catch (error) {
+        console.error('Error while zooming in:', error);
+      }
+    }
+  };
+
+  // Zoom Out Handler
+  const handleZoomOut = () => {
+    if (chart && chart.seriesContainer) {
+      try {
+        const currentZoom = chart.get('zoomLevel', 1);
+        const minZoom = chart.get('minZoomLevel', 1);
+
+        const newZoom = Math.max(currentZoom / 2, minZoom);
+
+        const centerX = chart.seriesContainer.width() / 2;
+        const centerY = chart.seriesContainer.height() / 2;
+
+        chart.zoomToPoint({x: centerX, y: centerY}, newZoom, true, 300);
+      } catch (error) {
+        console.error('Error while zooming out:', error);
+      }
+    }
+  };
+
+  // Reset view
+  const handleReset = () => {
+    if (chart) {
+      try {
+        chart.goHome(300);
+        onHomeClick?.(); // Callback to HeatMap.tsx to synch selected area
+      } catch (error) {
+        console.error('Error while resetting zoom:', error);
+      }
+    }
+  };
```

Comment:
Please wrap the callbacks in a `useCallback` for now.

Reply:
wrapped in useCallback.

https://github.com/DLR-SC/ESID/pull/384/commits/9f9a658e46c5e7673927bd331d99dd6d6542bf53